### DISCUSSION TITLE
Adding element/icon to title

### DISCUSSION THREAD

**Discussion by mehmetpekcan** [mathuo/dockview#423] [Ideas]
Hey, I was looking for adding an icon to my panel title, and I realized that it is already possible to add icons to `panel.title` even if its type is `string | undefined`. I was just thinking if it is best way to add icon to title. 

```js
event.api.addPanel({
  id: 'panel_1',
  component: 'panel_1',
  title: (
    <div>
      <AnyIcon />
      Test
    </div>
  ), 
});
```

I also use `localStorage` to keep user preferences, and it causes rendering problem

**Comment by techbech** [comment]
Didn't even know that was possible. I think the intended architecture would suggest to use custom tab title components for such things as described here https://dockview.dev/docs/components/dockview#custom-tab-title. At least what I'm doing :slightly_smiling_face:

**Comment by mathuo** [comment]
There are currently two ways to alter the behaviour of the header tabs. You can provide an alternative default template which will be used instead of the provided one or you can provide a template per panel when adding those panels.

Both examples can be found here:
https://dockview.dev/docs/next/components/dockview#custom-tab-headers

Does this help for your case?

**Comment by mehmetpekcan** [comment]
Yes, it definitely worked! I read the docs more than 10 times, but still it is a little bit confusing for me to understand some of them without digging. Thanks in advance @techbech @mathuo

**Comment by bhomoki** [comment]
Now that a newer version doesn't support this and the above mentioned docs links are dead, how to implement this feature? I'd like to add a jsx icon to each tabs, before the title string. Any solutions to that? Thanks!

**Reply by mathuo** [reply]
I would recommend checking out a custom tab implementation https://dockview.dev/docs/core/panels/tabs#extend-the-default-tab-implementation

Let me know if that's useful

**Reply by bhomoki** [reply]
Thanks @mathuo for the response. I tried that, could add an icon but had 2 issues with it.
1. the tab close button was gone for good and I needed that
2. when storing the user's config in local storage, the icon (or any reference) and some other related info is simply not stored and it caused issues so I gave up on this.
Also, the docs have many page-not-found CodeSandbox references which makes you feel that an older version had these features but the current one does not.

**Reply by lawrencenika** [reply]
any other methods? I need the same feature for allowing adding icon as prefix to the tab title here. but the custom tab implementation doc's codesandbox is already showing not found.

**Reply by husayt** [reply]
One could have bespoke save and load layout functionality while custom tab serialization is not there. I have done that before and works like a charm. To be honest, how do you expect your custom tab with icons to be serialised automatically. So custom logic is the way
